### PR TITLE
Singular data tests can now be documented

### LINKED ISSUE #6132: [Core] Singular data tests can now be documented
At a minimum, update body to include a link to the page on docs.getdbt.com requiring updates and what part(s) of the page you would like to see updated.
Originating from this issue: https://github.com/dbt-labs/dbt-core/issues/9005
~Original pull request: https://github.com/dbt-labs/dbt-core/pull/10744~
New pull request: https://github.com/dbt-labs/dbt-core/pull/10792

## Page(s) to update

- https://docs.getdbt.com/reference/resource-properties/description (add new "Data tests" tab)
- https://docs.getdbt.com/docs/build/data-tests (maybe 🤷)
- https://docs.getdbt.com/reference/project-configs/docs-paths (since it now includes [`test-paths`](https://docs.getdbt.com/reference/project-configs/test-paths))

## Code examples

`tests/my_singular_test.sql`

```sql
with my_cte as (
    select 1 as id, 'foo' as name
    union all
    select 2 as id, 'bar' as name
)
select * from my_cte
```

### Without docs blocks

`tests/_properties.yml`

```yaml
data_tests:
  - name: my_singular_test
    description: Some docs
    config:
      error_if: ">10"
    meta:
      some_key: some_val
```

### With docs blocks

Or with docs blocks:

`tests/example.md`

```md
{% docs my_singular_test_documentation %}
Some docs from a doc block
{% enddocs %}
```

`tests/_properties.yml`

```yaml
data_tests:
  - name: my_singular_test
    description: "{{ doc('my_singular_test_documentation') }}"
    config:
      error_if: ">10"
    meta:
      some_key: some_val
```

### List 'em and run 'em

List 'em:

```shell
$ dbt list --resource-type test -q 

my_project.my_singular_test
```

Run 'em:

```shell
$ dbt test -s my_project.my_singular_test

19:55:23  Running with dbt=1.9.0-a1
19:55:23  Registered adapter: postgres=1.9.0-a1
19:55:23  Found 1 test, 423 macros
19:55:23  
19:55:23  Concurrency: 5 threads (target='postgres')
19:55:23  
19:55:24  1 of 1 START test my_singular_test ............................................. [RUN]
19:55:24  1 of 1 FAIL 2 my_singular_test ................................................. [FAIL 2 in 0.10s]
19:55:24  
19:55:24  Finished running 1 test in 0 hours 0 minutes and 1.13 seconds (1.13s).
19:55:24  
19:55:24  Completed with 1 error, 0 partial successs, and 0 warnings:
19:55:24  
19:55:24  Failure in test my_singular_test (tests/my_singular_test.sql)
19:55:24    Got 2 results, configured to fail if != 0
19:55:24  
19:55:24    compiled code at target/compiled/my_project/tests/my_singular_test.sql
19:55:24  
19:55:24  Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
```

### PR DESCRIPTION
**Pull request by dbeatty10**
Resolves https://github.com/dbt-labs/docs.getdbt.com/issues/6132
Originally https://github.com/dbt-labs/docs.getdbt.com/pull/6147

## What are you changing in this pull request and why?

The `description` property is available for singular data tests beginning in dbt v1.9, and this PR highlights how singular tests can now be documented.

### 🎩 Previews
- [`description`](https://docs-getdbt-com-git-dbeatty-6132-docs-dbt-labs.vercel.app/reference/resource-properties/description)
- [`docs-paths`](https://docs-getdbt-com-git-dbeatty-6132-docs-dbt-labs.vercel.app/reference/project-configs/docs-paths)
- [documentation placement](https://docs-getdbt-com-git-dbeatty-6132-docs-dbt-labs.vercel.app/docs/build/documentation#placement)

### Additional information

Opened https://github.com/dbt-labs/docs.getdbt.com/issues/6193 since we don't have documentation how to add a `description` for the Jinja macro associated with custom generic tests. This follow-up will allow us to consider the `description` of both generic and singular data tests in context of each other.

## Checklist
- [x] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [x] Wait until https://github.com/dbt-labs/dbt-core/pull/10792 is merged
- [x] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)

### CODE REVIEW

File: website/docs/reference/resource-properties/description.md [hunk 1]
```diff
@@ -145,6 +146,33 @@ macros:
...
+
+```
+
+</File>
+
+</VersionBlock>
+
+<VersionBlock lastVersion="1.8">
+
+The `description` property is available for [singular data tests](/build/data-tests#document-singular-data-tests) beginning in dbt v1.9.
```
Comment (mirnawong1) [inline website/docs/reference/resource-properties/description.md]: @dbeatty10 this is linking to a page not found -- shouldn't this pr include a 'document singular data tests' header in the 'documentation' page for top level keys?

Comment (dbeatty10) [reply website/docs/reference/resource-properties/description.md]: Good catch! I'll fix this

Comment (mirnawong1) [inline website/docs/reference/resource-properties/description.md]: removing link
```suggestion
The `description` property is available for singular data tests beginning in dbt v1.9.
```

**Review by mirnawong1 — APPROVED** [review]
small tweak!

File: website/docs/reference/resource-properties/description.md (hunk 1 above)
Comment (dbeatty10) [reply website/docs/reference/resource-properties/description.md]: Thx for fixing this! 💪

**Comment by graciegoheen** [comment]
@dbeatty10 @mirnawong1 I'm realizing this doesn't have a release note - should we add one?

**Comment by mirnawong1** [comment]
good point and have created a release note PR here: https://github.com/dbt-labs/docs.getdbt.com/pull/6227

**Pull request merged**